src/hypervisor.c

```diff
@@ -107,6 +108,43 @@ cc_oci_expand_net_device_cmdline(struct cc_oci_config *config, guint index) {
 	return g_strdup("");
 }
 
+/*
+ *
+ *
+ */
+#define DPDK_CMDLINE_OBJ "memory-backend-file,id=dpdkmem,size=2048M,mem-path=/dev/hugepages,share=on,prealloc=on"
```

Comment:
I think we're potentially talking about two different things here. The relevant snippet from `hypervisor.args.in`:

```
# XXX: memory needed to map clear-containers.img into the guest is calculated dynamically
# XXX: based on the size of the image.
-object
memory-backend-file,id=mem0,mem-path=@IMAGE@,size=@SIZE@

# XXX: memory assigned to the VM is hard-coded currently.
-m
2G,slots=2,maxmem=3G
```
How likely is it that users will want/need to modify the memory size specified in `DPDK_CMDLINE_OBJ`? Since dpdk is optional, that value cannot be specified in `hypervisor.args`, but having to rebuild the runtime to change the value is pretty horrid (and also stops users from consuming the packaged version of the runtime along with the potential security implications of that).

If we think that value might need to change, how about we provide a way to override this hard-coded default via an additional argument that the runtime can accept? There is already precedent here:

  - `--hypervisor-log-dir=`
  - `--shim-path=`
  - `--proxy-socket-path=`

Once that argument exists, the only extra requirement is that docker be configured to use `/usr/bin/cc-oci-runtime.sh` and then `--dpdk-mem-size=` could be added to `/etc/cc-oci-runtime/cc-oci-runtime.sh.cfg`.